db/seeds.rb

```diff
@@ -1 +1,100 @@
-load(Rails.root.join("db", "seeds", "#{Rails.env.downcase}.rb"))
+# This seed script populates the development DB with a data set whose size is dependent on the Rails environment.
+
+# You can control the randomness of the data provided by FAKER and the Rails libraries via the DB_SEEDS_RANDOM_SEED environment variable.
+# If you specify a number, that number will be used as the seed, so you can enforce consistent data across runs
+#   with nondefault content.
+# If you specify the string 'random' (e.g. `export DB_SEEDS_RANDOM_SEED=random`), a random seed will be assigned for you.
+# If you don't specify anything, 0 will be used as the seed, ensuring consistent data across hosts and runs.
+
+require "faker"
+require_relative 'seeds/casa_org_populator_presets'
+require_relative 'seeds/db_populator'
+require_relative '../lib/tasks/data_post_processors/case_contact_populator'
+require_relative '../lib/tasks/data_post_processors/contact_type_populator'
+
+class SeederMain
+
+  attr_reader :db_populator, :rng
+
+  def initialize
+    random_seed = get_seed_specification
+    @rng = Random.new(random_seed)  # rng = random number generator
+    @db_populator = DbPopulator.new(rng)
+    Faker::Config.random = rng
+  end
+
+  def seed
+    puts "Erasing all objects from the data base..."
+    destroy_all
+
+    puts "Creating the objects in the data base..."
+    db_populator.create_all_casa_admin('allcasaadmin@example.com')
+    db_populator.create_org(CasaOrgPopulatorPresets.for_environment.merge({org_name: 'Prince George CASA'}))
+    db_populator.create_org(CasaOrgPopulatorPresets.minimal_dataset_options)
+
+    post_process_data
+
+    report_object_counts
+    puts "\nDone.\n\n"
+  end
+
+  private # -------------------------------------------------------------------------------------------------------
+
+  # Used for reporting record counts after completion:
+  def active_record_classes
+    @active_record_classes ||= [
+        CasaOrg,
+        CasaCase,
+        User,
+        Volunteer,
+        Supervisor,
+        CasaAdmin,
+        AllCasaAdmin,
+        SupervisorVolunteer,
+        CaseAssignment,
+        ContactType,
+        ContactTypeGroup,
+        CaseContact,
+    ]
+  end
+
+  def destroy_all
+    # Order is important here; CaseContact must be destroyed before the User that created it.
+    # The User is destroyed as a result of destroying the CasaOrg.
+    [SupervisorVolunteer, CaseContact, CasaOrg, AllCasaAdmin, ContactTypeGroup, ContactType].each { |klass| klass.destroy_all }
+    non_empty_classes = active_record_classes.select { |klass| klass.count > 0 }
+    unless non_empty_classes.empty?
```

Comment:
I think an`if` statements might be a little easier to read here:
```
if non_empty_classes.any?
```

Reply:
I agree with you, that would be easier to understand.